### PR TITLE
Fix for management port speed issue

### PR DESCRIPTION
**Pull request by sumukhatv**
**- What I did**
Fixed the SNMP port speed missing issue.
**- How I did it**
Port speed was not being updated in the Config DB. Made changes to the minigraph parser to obtain the management port speed information from the minigraph. Now that the parser will return this information to the config-engine, which will be updated in the Config DB.

**- How to verify it**
Check the config DB for the changes in the MGMT_PORT key. It should then contain speed value as well.
**- Description for the changelog**

I added the logic to lookup for the ManagementInterface section in the minigraph xml file and add this to the dictionary object returned by parse_xml method in the minigraph.py module

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -484,6 +493,7 @@ def parse_xml(filename, platform=None, port_config_file=None):
     results['MGMT_PORT'] = {}
     results['MGMT_INTERFACE'] = {}
     mgmt_intf_count = 0
+    mgmt_speed = 0
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py]: >mgmt_speed [](start = 4, length = 10)

not used

File: src/sonic-config-engine/minigraph.py [hunk 2]
```diff
@@ -403,6 +403,15 @@ def parse_deviceinfo(meta, hwsku):
...
                 port_speeds[port_alias_map.get(alias, alias)] = speed
+            interfaces = device_info.find(str(QName(ns, "ManagementInterfaces")))
+            if interfaces:
+                for interface in interfaces.findall(str(QName(ns1, "ManagementInterface"))):
+                    alias = interface.find(str(QName(ns, "InterfaceName"))).text
+                    speed = interface.find(str(QName(ns, "Speed"))).text
+                    desc  = interface.find(str(QName(ns, "Description")))
+                    if desc != None:
+                        port_descriptions[port_alias_map.get(alias, alias)] = desc.text
+                    port_speeds[port_alias_map.get(alias, alias)] = speed
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py]: You may combine 2 loops into one.

File: src/sonic-config-engine/minigraph.py [hunk 3]
```diff
@@ -493,7 +503,11 @@ def parse_xml(filename, platform=None, port_config_file=None):
             name = 'eth' + str(mgmt_intf_count)
             mgmt_intf_count += 1
             mgmt_alias_reverse_mapping[alias] = name
-        results['MGMT_PORT'][name] = {'alias': alias, 'admin_status': 'up'}
+        if port_speeds_default.has_key(alias):
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py]: >port_speeds_default.has_key(alias) [](start = 11, length = 34)

alias in port_speeds_default

File: src/sonic-config-engine/minigraph.py [hunk 4]
```diff
@@ -493,7 +503,11 @@ def parse_xml(filename, platform=None, port_config_file=None):
...
+            port_speed = port_speeds_default[alias]
+            results['MGMT_PORT'][name] = {'alias': alias, 'admin_status': 'up', 'speed': port_speed}
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py]: >'speed': port_speed} [](start = 80, length = 20)

suggest append it conditionally. if we have more optional field, the if-else doesn't work well.

**Comment by qiluo-msft** [comment]
> #!/usr/bin/env python

Where is the unit test?

---
Refers to: src/sonic-config-engine/minigraph.py:1 in d1f04a2. [](commit_id = d1f04a2986b6f06df8c42a4a43ab8016f39849f3, deletion_comment = False)

File: src/sonic-config-engine/minigraph.py (hunk 2 above)
Comment (sumukhatv) [reply src/sonic-config-engine/minigraph.py]: Which is the other loop you are referring to?

**Pull request merged**